pandas/core/frame.py

```diff
@@ -2212,6 +2212,63 @@ def insert(self, loc, column, value, allow_duplicates=False):
         self._data.insert(
             loc, column, value, allow_duplicates=allow_duplicates)
 
+    def assign(self, **kwargs):
+        """
+        Assign new columns to a DataFrame.
+
+        Parameters
+        ==========
+
+        **kwargs : keyword, value pairs
+            keywords are the column names, and values are either
+            inserted into the DataFrame, or called on the DataFrame
+            and inserted if the value is callable.
+
+        Returns
+        =======
+
+        df : DataFrame
+            A new DataFrame with the new columns in addition to
+            all the existing columns.
+
+        Notes
+        =====
+
+        Since **kwargs is essentially dictionary, the order of your
+        arguments may not be preserved, and so the order of the
+        resulting columns will be undefined. While assigning multiple
+        columns within the same ``assign`` is possible, you cannot
+        reference other columns created within the same ``assign`` call.
+
+        Examples
+        ========
+
+        df = DataFrame({'A': range(1, 10), 'B': np.random.randn(10)})
```

Comment:
yes, indeed, blocks that start with `>>>` are automatically converted to code blocks